framework/web/Request.php

```diff
@@ -162,8 +162,51 @@ class Request extends \yii\base\Request
      * @see getBodyParams()
      */
     public $parsers = [];
+    /**
+     * @var array The configuration for trusted security related headers.
+     * An array key is a regular expression for matching a hostname.
+     * A value is a list of headers to trust.
+     *
+     * For example, to trust all headers from domains ending in '.trusted.com' use the following:
+     *
+     * ```php
+     * [
+     *     '/^.*\.trusted\.com$/',
+     * ```
+     *
+     * To trust just the x-forwarded-for header from domains ending in '.partial.com' use:
+     *
+     * ```
+     *     // ...
+     *     '/^.*\.partial\.com$/' => ['X-Forwarded-For']
+     * ]
+     * ```
+
+     * Default is to trusts all headers from all hosts.
+     */
+    public $trustedHostConfig = [
+        '//'
+    ];
+    /**
+     * Lists headers that are subject to the trusted host configuration.
+     * @see https://en.wikipedia.org/wiki/List_of_HTTP_header_fields
+     * @var array
+     */
+    public $secureHeaders = [
+        'X-Forwarded-For',
+        'X-Forwarded-Host',
+        'X-Forwarded-Proto',
+        'Front-End-Https',
+    ];
 
     /**
+     * @var string[] List of headers where proxies store the real client IP, it's not advisable to put untrusted headers
```

Comment:
Note that @rob006 made good arguments in favor of using `fnmatch` instead of `preg_match` for matching host names / ips.
What's your opinion?